main.py

```diff
@@ -17,28 +27,183 @@ class YAMLParam(TypedDict):
+        else:
+            units_map[param] = "dimensionless"
+    
+    return units_map
+
+def create_graph(df: pd.DataFrame, param: str, param_unit: str, output_path: str):
+  
+    plt.style.use('seaborn-v0_8-darkgrid')
+    
+    fig, ax = plt.subplots(figsize=(12, 6), dpi=300)
+ 
+    times = pd.to_datetime(df['datetime'])
+    
+    # Plot the parameter
+    ax.plot(times, df[param], linewidth=2.5, marker='o', 
+           markersize=4, markevery=max(1, len(df) // 50), color="#B923AA")
+    
+    ax.xaxis.set_major_formatter(mdates.DateFormatter('%H:%M'))
+    ax.xaxis.set_major_locator(mdates.HourLocator(interval=1))
+    plt.setp(ax.xaxis.get_majorticklabels(), rotation=45, ha='right')
+    
+    ax.set_xlabel('Time', fontsize=12, fontweight='bold')
+    
+    # Format y-axis label with parameter name and unit
+    if param_unit != "dimensionless":
+        ylabel = f"{param} ({param_unit})"
+    else:
+        ylabel = param
+    ax.set_ylabel(ylabel, fontsize=12, fontweight='bold')
+    
+    # Title with parameter name
+    ax.set_title(f'{param} Over Time', fontsize=14, fontweight='bold', pad=20)
+    
+    ax.grid(True, alpha=0.3, linestyle='--', linewidth=0.5)
+    
+    plt.tight_layout()
+    
+    # Save the figure
+    plt.savefig(output_path, bbox_inches='tight', facecolor='white', edgecolor='none')
+    plt.close()
+    
+    print(f"Graph saved to {output_path}")
+
+def generate_graphs(df: pd.DataFrame, graph_params: list[str], units_map: dict[str, str], output_dir: str):
+   
+    output_path = Path(output_dir)
+    output_path.mkdir(parents=True, exist_ok=True)
+  
+    missing_params = [p for p in graph_params if p not in df.columns]
+    if missing_params:
+        print(f"The parameters were not found in the data: {missing_params}")
+        graph_params = [p for p in graph_params if p in df.columns]
+    
+    if not graph_params:
+        print("No valid parameters to graph.")
+        return
+    
+    # Create a separate graph for each parameter
+    for param in graph_params:
+        # Create filename based on parameter name
+        filename = f"{param}_graph.png"
+        output_file = output_path / filename
+        
+        # Get unit for this parameter
+        param_unit = units_map.get(param, "dimensionless")
+        
+        create_graph(df, param, param_unit, str(output_file))
+
 def main():
+    # Command-line arguments
+    parser = argparse.ArgumentParser(description="Run vehicle model and log parameters.")
+    parser.add_argument(
+        "--log",
+        nargs="+",
+        help="List of parameter names to log each timestep",
+        required=True
+    )
+    parser.add_argument(
+        "--csv",
+        default="log.csv",
+        help="Output CSV filename (default: log.csv)"
+    )
+    parser.add_argument(
+        "--graph",
+        nargs="+",
+        help="List of parameter names to graph over time",
+        default=None
+    )
```

Comment:
Maybe we could just decide to produce a graph for every logged parameter by default, if --graph is not specified. This would make the command easier so we don't have to type all the same params twice.